src/AsyncTestCase.php

```diff
@@ -55,11 +56,13 @@ abstract class AsyncTestCase extends PHPUnitTestCase
         $start = \microtime(true);
 
         Loop::run(function () use (&$returnValue, &$exception, &$invoked, $args) {
+            yield $this->setUpAsync();
             $promise = call([$this, $this->realTestName], ...$args);
             $promise->onResolve(function ($error, $value) use (&$invoked, &$exception, &$returnValue) {
                 $invoked = true;
                 $exception = $error;
                 $returnValue = $value;
+                yield $this->tearDownAsync();
```

Comment:
There shouldn't be a yield inside onResolve.

Reply:
Why not? It seems to be supported: https://github.com/amphp/amp/blob/c34b6791429c0c55d8b66df748a1c265d6b9e75a/lib/Internal/Placeholder.php#L45-L54